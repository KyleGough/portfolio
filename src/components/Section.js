import React from 'react';

export default function Section(props) {
    return (
        <div className='container text-primary py-8 last:pb-16'>
            {props.children}
        </div>
    );
}