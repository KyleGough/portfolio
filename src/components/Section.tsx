import { ReactNode } from 'react';

export default function Section(props: { children: ReactNode }) {
    return (
        <div className='container text-primary py-8 last:pb-16'>
            {props.children}
        </div>
    );
}