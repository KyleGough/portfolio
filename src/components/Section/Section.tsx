import { clsx } from 'clsx';
import React, { forwardRef } from 'react';

interface SectionProps {
  id?: string;
  overrideTopPadding?: boolean;
  children: React.ReactNode;
}

const Section = forwardRef<HTMLDivElement, SectionProps>(
  ({ id, overrideTopPadding = false, children }, ref) => (
    <div
      id={id}
      ref={ref}
      className={clsx(
        { 'first:pt-24': overrideTopPadding },
        'container text-primary py-8 last:pb-16'
      )}
    >
      {children}
    </div>
  )
);

Section.displayName = 'Section';
export { Section };
