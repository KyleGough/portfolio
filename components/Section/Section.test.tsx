import { render, screen } from '@testing-library/react';
import React from 'react';

import { Section } from './Section';

describe('Section component', () => {
  it('renders', async () => {
    const { container } = render(<Section>Test</Section>);
    expect(screen.getByText('Test')).toBeVisible();
    expect(container).toMatchSnapshot();
  });

  it('component overrides top padding', async () => {
    render(
      <Section id="test" overrideTopPadding={true}>
        Test
      </Section>
    );

    const section = screen.getByText('Test');
    expect(section.className.includes('first:pt-24')).toBe(false);
  });
});
