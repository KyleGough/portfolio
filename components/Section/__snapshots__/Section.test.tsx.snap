// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    Test
  </div>
</div>
`;
